### PR TITLE
[BB-6370] feat: add instance configuration settings to grove payload

### PR DESCRIPTION
**Pull request by kaustavb12**
## Description

This PR adds support for including the following configuration settings to the grove trigger payload, while triggering a new deployment:

1. Theme repo information and theme override settings
2. Site configuration settings
3. LMS and CMS env configuration settings

## Supporting information

[BB-6370](https://tasks.opencraft.com/browse/BB-6370)

## Testing instructions

1. Checkout this branch
2. Add the grove related configuration to the .env.test file
3. Setup local environment using [these instructions](https://ocim.opencraft.com/en/latest/development/docker/)
4. From the django shell, create a new instance using [these instructions](https://ocim.opencraft.com/en/latest/provisioning-sandboxes/)
5. Go to [console ui](http://localhost:3000/) and signup. This should create a new BetaTestApplication.
6. Go to the [django admin page](http://localhost:5000/admin/registration/betatestapplication/), and add the instance id of the Grove instance and instance type as `grove instance` for the newly create BetaTestApplication
7. Go to your Grove instance in the [django admin panel](http://localhost:5000/admin/grove/groveinstance/).
8. Enable `Deploy simple_theme` checkbox
9. Add sample theme settings such as this :
```json
{"version": 1, "footer-bg": "#FFFFFF", "header-bg": "#FFFFFF", "link-color": "#126F9A", "main-color": "#126F9A", "btn-primary-bg": "#126F9A", "btn-secondary-bg": "#FFFFFF", "btn-primary-color": "#FFFFFF", "btn-secondary-color": "#126F9A", "btn-primary-hover-bg": "#FFFFFF", "btn-secondary-hover-bg": "#126F9A", "btn-primary-hover-color": "#126F9A", "btn-primary-border-color": "#126F9A", "btn-secondary-hover-color": "#FFFFFF", "btn-secondary-border-color": "#126F9A", "home-page-hero-title-color": "#313131", "home-page-hero-subtitle-color": "#646464", "btn-primary-hover-border-color": "#126F9A", "btn-secondary-hover-border-color": "#FFFFFF"}
```
10. Add sample static content in the `Static content overrides JSON` field as this one:
```json
{"version": 0, "homepage_overlay_html": "<h1>Welcome to Kaustav Testing</h1><p>It works! Powered by Open edX\u00ae</p>"}
```
11. Add sample env configuration in the `Configuration extra settings` field such as this one:
```yaml
EDXAPP_LMS_ENV: 
    DCS_SESSION_COOKIE_SAMESITE: Lax
EDXAPP_LMS_ENV_FEATURES:
    ENABLE_CREATOR_GROUP: true
EDXAPP_CMS_ENV:
    SESSION_ENGINE: django.contrib.sessions.backends.cached_db
EDXAPP_CMS_ENV_FEATURES:
    ENABLE_BULK_ENROLLMENT_VIEW: true
EDXAPP_COMMON_ENV_FEATURES:
    ENABLE_COUNTRY_ACCESS: true
EDXAPP_LOG_LEVEL: info
EDXAPP_PDF_RECEIPT_TAX_ID: TAX123
EDXAPP_CMS_STATIC_URL_BASE: http://static.example.com
EDXAPP_ENABLE_MKTG_SITE: true
```
12. From the [console backend UI](http://localhost:5000/instance/), trigger a new deployment
13. Go the the gitlab pipeline for the configured Grove repo, and verify the payload as well as the resulting configuration of the instances.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.

### CODE REVIEW

File: instance/tests/api/test_openedx_appserver.py
Comment (gabor-boros) [inline instance/tests/api/test_openedx_appserver.py:274]: 👍 I thought that I already did this flip. -- Maybe I did it on the GitLab fork? 🤔

File: instance/tests/models/test_openedx_database_mixins.py
Comment (kaustavb12) [reply instance/tests/models/test_openedx_database_mixins.py:859]: This particular test case was failing, since the rabbitMQ settings were no longer available after the switch.
Should I remove it completely ?

Comment (gabor-boros) [reply instance/tests/models/test_openedx_database_mixins.py:859]: @kaustavb12 I think removing it makes sense. We won't fallback to RabbitMQ.

Comment (ngdangdat) [reply instance/tests/models/test_openedx_database_mixins.py]: Hi @kaustavb12,

This test case is blocking my PR here: https://github.com/open-craft/opencraft/pull/907. Will we merge this PR any time soon this sprint? Or can we have this one fixed in a separated PR? Either is good for me.

My change is not critical but this is the last sprint of my trial :D

----

By the way, I wonder how you get tests passed for this case: https://github.com/open-craft/opencraft/blob/master/instance/tests/models/test_database_server.py#L298-L302 -- MySQL information is used for MongoDB assertion and it fails in my branch.

Comment (kaustavb12) [reply instance/tests/models/test_openedx_database_mixins.py]: @ngdangdat 

Currently there are two things blocking the merge of the PR:
1. The final integration testing of this PR with Grove is blocked by [Grove!89](https://gitlab.com/opencraft/dev/grove/-/merge_requests/89), since that MR is a major overhaul of the Grove platform and it makes sense to test it once that is merged and released. As of writing of this update, that MR has just been merged, so we should be good to go for testing here any day now.
2. #903 - This is a high priority PR and I would prefer to hold any merge to OCIM till after that is merged and deployed. Currently that PR is ready to be deployed and is blocked by ongoing mass deployments in OCIM, which should be over in couple of days.

Coming to the test cases. #903 also fixes the test cases involving RabbitMQ, so you should be good to go after that is merged.

> By the way, I wonder how you get tests passed for this case: https://github.com/open-craft/opencraft/blob/master/instance/tests/models/test_database_server.py#L298-L302 -- MySQL information is used for MongoDB assertion and it fails in my branch.

That test case is passing by default. I did not have to do anything for that. Also I _did not_ see that test case [failing for your PR](https://app.circleci.com/pipelines/github/open-craft/opencraft/4939/workflows/daac1c5d-cd0a-4655-b777-33f5092383a8/jobs/42783). Not sure if I am missing something here.

Comment (ngdangdat) [reply instance/tests/models/test_openedx_database_mixins.py]: @kaustavb12 Well noted. Thanks for the information!

> That test case is passing by default. I did not have to do anything for that. Also I did not see that test case [failing for your PR](https://app.circleci.com/pipelines/github/open-craft/opencraft/4939/workflows/daac1c5d-cd0a-4655-b777-33f5092383a8/jobs/42783). Not sure if I am missing something here.

You're right! Sorry for not mentioning the environment. I triggered running test in my local environment (Docker one).

**Review by gabor-boros — APPROVED** [review]
👍  🎉 @kaustavb12 Grove deployments are broken due to some ongoing work, but I can confirm that the config is passed and rendered correctly.

- [x] I tested what's in the test instructions
- [x] I read through the code

**Pull request merged**
